data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -317,24 +337,17 @@ then
 	touch cco-bootstrap.done
 fi
 
-# Wait for the etcd cluster to come up.
-until bootkube_podman_run \
-		--rm \
-		--name etcdctl \
-		--env ETCDCTL_API=3 \
-		--volume /opt/openshift/tls:/opt/openshift/tls:ro,z \
-		--entrypoint etcdctl \
-		"${MACHINE_CONFIG_ETCD_IMAGE}" \
-		--dial-timeout=10m \
-		--cacert=/opt/openshift/tls/etcd-ca-bundle.crt \
-		--cert=/opt/openshift/tls/etcd-client.crt \
-		--key=/opt/openshift/tls/etcd-client.key \
-		--endpoints="${ETCD_ENDPOINTS}" \
-		endpoint health
-do
-	echo "etcdctl failed. Retrying in 5 seconds..."
-	sleep 5
-done
+# in case of single node, if we removed etcd, there is no point to wait for it on restart
```

Comment:
Would it be better to have the logic of the script stay the same in all forms, but to generate settings in variables that are different when we're using the bootstrap-in-place approach? That would benefit someone debugging multiple clusters of different types, since the script content would always be the same.

For example, if we put something like this early in the file

```
{{- if .BootstrapInPlace }}
BOOTSTRAP_INPLACE=true
{{ else }}
BOOTSTRAP_INPLACE=false
{{ end -}}
```

then, in the places where it matters, the shell script can test that variable.

Reply:
We can definitely add this variable or a comment saying `###USING BOOTSTRAP IN PLACE CONFIGURATION`.
 Just note that this is a template and we are switching on {{- if .BootstrapInPlace}} at script-creation-time and not run-time.

Reply:
Right, I get that this is a template. My point is that we shouldn't be changing the _logic_ of the script at compile time, we should only change the inputs. Changing the logic means that someone looking at the raw input file has to run it through a golang template translator in their head to understand which lines should appear on a single-node vs. multi-node deployment for debugging, instead of being able to follow the shell logic after seeing that the template simply sets a variable near the top of the file.

Maybe the other reviewers are less concerned with that new style of change in this file, though, since no one else has mentioned it.

Reply:
For what it's worth, I made a similar point on one of my earlier [reviews](https://github.com/staebler). I am in favor of preferring script conditional logic over template conditional logic.

Reply:
OK, I don't have a preference for one over the other, I'll update the script.